client/network_connect_test.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/docker/engine-api/client/transport"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/network"
+)
+
+func TestNetworkConnectError(t *testing.T) {
+	client := &Client{
+		transport: transport.NewMockClient(nil, transport.ErrorMock(http.StatusInternalServerError, "Server error")),
+	}
+
+	err := client.NetworkConnect("network_id", "container_id", nil)
+	if err == nil || err.Error() != "Error response from daemon: Server error" {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestNetworkConnect(t *testing.T) {
+	expectedURL := "/networks/network_id/connect"
+
+	client := &Client{
+		transport: transport.NewMockClient(nil, func(req *http.Request) (*http.Response, error) {
+			if !strings.HasPrefix(req.URL.Path, expectedURL) {
+				return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
+			}
+
+			if req.Method != "POST" {
+				return nil, fmt.Errorf("expected POST method, got %s", req.Method)
+			}
+
+			var connect types.NetworkConnect
+			if err := json.NewDecoder(req.Body).Decode(&connect); err != nil {
+				return nil, err
+			}
+
+			if connect.Container != "container_id" {
+				return nil, fmt.Errorf("expected 'container_id', got %s", connect.Container)
+			}
+
+			if connect.EndpointConfig.NetworkID != "NetworkID" {
+				return nil, fmt.Errorf("expected 'NetworkID', got %s", connect.EndpointConfig.NetworkID)
+			}
+
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       ioutil.NopCloser(bytes.NewReader([]byte(""))),
+			}, nil
+		}),
+	}
+
+	err := client.NetworkConnect("network_id", "container_id", &network.EndpointSettings{
```

Comment:
Should we test for cases where the EndpointConfig is nil to ensure that we don't see anything on the wire?